channel.go

```diff
@@ -427,7 +441,7 @@ func (ch *Channel) Connect(ctx context.Context, hostPort string, connectionOptio
 // incomingConnectionActive adds a new active connection to our peer list.
 func (ch *Channel) incomingConnectionActive(c *Connection) {
 	c.log.Debugf("Add connection as an active peer for %v", c.remotePeerInfo.HostPort)
-	p := ch.peers.GetOrAdd(c.remotePeerInfo.HostPort)
+	p := ch.rootPeers().GetOrAdd(c.remotePeerInfo.HostPort)
```

Comment:
So one behaviour we need to support is:
 - client adds one peer: "hyperbahn-1:21300", and then advertises with Hyperbahn
 - Hyperbahn then connects back to us using 10 connections
 - When client makes calls to Hyperbahn, it should be able to use those 10 connections (without having to manually add peers)

When we get an incoming connection, we need to wait for it to get identified (which is when we find out what service name it is), and probably add it to the right subchannel's peer lists based on the service name.

Let's add a test for that, and see if it works the way we expect.